typing/ctype.ml

```diff
@@ -6073,10 +6069,10 @@ let () =
   Env.same_constr := same_constr
 
 let is_immediate env ty =
-  let layout : Type_layout.t =
+  let layout =
     (* In bytecode, we don't know at compile time whether we are
        targeting 32 or 64 bits. *)
-    if !Clflags.native_code && Sys.word_size = 64 then Immediate64
-    else Immediate
+    if !Clflags.native_code && Sys.word_size = 64 then Layout.Immediate64
+    else Layout.Immediate
   in
-  Result.is_ok (check_type_layout env ty layout)
+  Result.is_ok (check_type_layout env ty (Layout.of_const layout))
```

Comment:
Minor nit:  Can you drop the `of_const` function call if you use replace `Immediate` with `immediate` and `Immediate64` with `immediate64` in the let?  (The original code also should have used the definitions from `Type_layout` rather than the constructors - my mistake)

Reply:
Yes, good point